File: server/OurCity.Api/Infrastructure/Database/Media.cs
using System;

namespace OurCity.Api.Infrastructure.Database;

public class Media
{
    public Guid Id { get; set; }
    public Guid PostId { get; set; }
    public string Url { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation property
    public Post? Post { get; set; }
}

File: webapp/src/types/dtos/media.ts
/*
  /webapp/src/models/dtos/media.ts

  This file contains the Data Transfer Objects (DTOs) related to the Media entity.
  The interfaces corresponding to each DTO are defined as per the request/response definitions in the API contract.
*/

export interface MediaResponseDto {
  id: string;
  postId: string;
  url: string;
  createdAt: string; // ISO date string
  updatedAt: string; // ISO date string
}